decoder/expr_any_hover.go

```diff
@@ -192,3 +195,34 @@ func (a Any) hoverOperatorExprAtPos(ctx context.Context, pos hcl.Pos) (*lang.Hov
 
 	return nil, false
 }
+
+func (a Any) hoverTemplateExprAtPos(ctx context.Context, pos hcl.Pos) (*lang.HoverData, bool) {
+	switch eType := a.expr.(type) {
+	case *hclsyntax.TemplateExpr:
+		if eType.IsStringLiteral() {
+			return nil, false
+		}
+
+		for _, partExpr := range eType.Parts {
+			if partExpr.Range().ContainsPos(pos) {
+				cons := schema.AnyExpression{
+					OfType: cty.String,
+				}
+				return newExpression(a.pathCtx, partExpr, cons).HoverAtPos(ctx, pos), true
```

Comment:
It looks like the literal strings still need to be accounted for when they're inside of HEREDOC here, because we end up in https://github.com/hashicorp/hcl-lang/blob/0c7d286e56219c3b3c05a7124ee8547b1e710bc2/decoder/expr_literal_type_hover.go#L27-L42

![2023-11-21 13 11 14](https://github.com/hashicorp/hcl-lang/assets/287584/66340c49-f787-45fb-ac72-5cc2254c89fc)